lib/cloud/azure/vm.go

```diff
@@ -166,3 +167,74 @@ func (c *vmClient) ListVirtualMachines(ctx context.Context, resourceGroup string
 
 	return virtualMachines, nil
 }
+
+// vmRunCommand provides an interface for an Azure Run Command client.
+type vmRunCommand interface {
```

Comment:
It looks like this interface might be unnecessary, see comments below.

https://github.com/golang/go/wiki/CodeReviewComments#interfaces